google/cloud/storage/client.py

```diff
@@ -112,6 +114,15 @@ def __init__(
         if project is _marker:
             project = None
 
+        if os.getenv(STORAGE_EMULATOR_ENV_VAR) is not None:
+            credentials = AnonymousCredentials()
+            if project in [_marker, None]:
```

Comment:
Yes, when we are passing `credentials` as a keyword argument and if not passing project then it found the project in fall back but then throw an error about Default credential from here https://github.com/googleapis/google-auth-library-python/blob/bc92abbce5e0a5e7b5de8157e7772db84349a6cb/google/auth/_default.py#L356 when called a method `google.auth.default()` method because method unable to find credential from  the given four options .

Reply:
I think i can remove this condition after merge of [cloud-core-51](https://github.com/googleapis/python-cloud-core/pull/51) PR